### PR TITLE
Added "Reopen In Container" explorer menu item

### PR DESCRIPTION
**Pull request by ChloeQuijano**
_(New PR - Fixed the files error)_ Added a new explorer menu item that reopens a local function app project in a dev container.
**"Reopens in Container"** appears when the user right clicks a folder that is NOT the `.devcontainer` folder. Otherwise, **"Update Dev Container"** appears when the user right clicks the `.devcontainer` folder, if it exists.

**"Reopens in Container"** - Reopens the project in a dev container
Screenshot 2021-08-02 115950

**"Update Dev Container"** - Rebuilds the .devcontainer folder and reopens the project in the dev container
Screenshot 2021-08-02 120040

### CODE REVIEW

File: src/commands/dockersupport/reopenInContainer.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        if (node) {
+            const message: string = localize('selectFunctionProject', 'Select your App Functions Project');
+            const projectFilePath: string = await getProjectFolder(context, message, context.workspacePath);
+            vscode.commands.executeCommand('remote-containers.reopenInContainer', vscode.Uri.file(projectFilePath));
+        } else {
+            void vscode.window.showInformationMessage(localize('noProjectOpened', 'Conditions not met. Cannot use Docker with this project'));
+        }
+    } catch (error) {
+        const message: string = localize('failReopenInContainer', 'Failed to reopen project in a dev container.');
+        throw new Error(message);
```
Comment (nertim) [inline src/commands/dockersupport/reopenInContainer.ts]: should we be throwing or showing an error message? it always scares me to see an exception being throw in client side apps.

File: src/commands/dockersupport/updateContainer.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * Rebuilds and reopens the opened function app project in a dev container
+*/
+export async function updateContainer(context: IActionContext, node: AzExtTreeItem): Promise<void> {
+
+    try {
+        const projectFilePath: string = await getProjectFolder(context, node.label);
+        vscode.commands.executeCommand('remote-containers.rebuildAndReopenInContainer', vscode.Uri.file(projectFilePath))
+    } catch (error) {
+        const message: string = localize('cannotRebuildandReopen', 'Failed to update and reopen project in a dev container.');
+        throw new Error(message);
```
Comment (nertim) [inline src/commands/dockersupport/updateContainer.ts]: similarly here.. should we throw or just show the error message to the user? also does the `error` object contain any info that could be useful to the user?

File: src/commands/dockersupport/reopenInContainer.ts (hunk 1 above)
Comment (ChloeQuijano) [reply src/commands/dockersupport/reopenInContainer.ts]: That's good to note, I never thought about it like that. I only need it to show an error, I will make the change.

File: src/commands/dockersupport/updateContainer.ts (hunk 2 above)
Comment (ChloeQuijano) [reply src/commands/dockersupport/updateContainer.ts]: Nope, the object doesn't contain any info the user needs. Will make the change

**Pull request closed without merging**